sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/SyncTokenPolicy.java

```diff
@@ -0,0 +1,100 @@
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * A policy uses a concurrent map to maintain the latest sync-tokens. When this HTTP pipeline policy is triggered, the
+ * policy will retrieve all sync-tokens without sequence number segment from the concurrent map and use it in the HTTP
+ * request. Also after received the HTTP response, update the latest sync-tokens to the map.
+ */
+public final class SyncTokenPolicy implements HttpPipelinePolicy {
+    private static final String COMMA = ",";
+    private static final String EQUAL = "=";
+    private static final String SYNC_TOKEN = "Sync-Token";
+    private final String SKIP_INVALID_TOKEN = "Skipping invalid sync token '{}'.";
+    private final Map<String, SyncToken> syncTokenMap = new ConcurrentHashMap<>(); // key is sync-token id
+    private final ClientLogger logger = new ClientLogger(SyncTokenPolicy.class);
+
+    /**
+     * Add or update the sync token to a thread safe map.
+     *
+     * @param context request context
+     * @param next The next policy to invoke.
+     * @return A {@link Mono} representing the HTTP response that will arrive asynchronously.
+     */
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+
+        // Add all of sync-tokens to HTTP request header
+        context.getHttpRequest().setHeader(SYNC_TOKEN, getSyncTokenHeader());
+
+        return next.process().flatMap(httpResponse -> {
+            // Get the sync-token from HTTP response header
+            final String syncTokenValue = httpResponse.getHeaders().getValue(SYNC_TOKEN);
+
+            // Skip sync-token updates of concurrent map if no 'Sync-Token' header
+            if (syncTokenValue != null) {
+                updateSyncToken(syncTokenValue);
+            }
+
+            return Mono.just(httpResponse);
+        });
+    }
+
+    /**
+     * Get all latest sync-tokens from the concurrent map and convert to one sync-token string.
+     * All sync-tokens concatenated by a comma delimiter.
+     * @return sync-token string
+     */
+    private String getSyncTokenHeader() {
+        return syncTokenMap.values().stream().map(syncToken -> syncToken.getId() + EQUAL + syncToken.getValue())
+                   .collect(Collectors.joining(COMMA));
+    }
+
+    /**
+     * Update the existing synchronization tokens.
+     *
+     * @param token an external synchronization token to ensure service requests receive up-to-date values.
+     */
+    public void updateSyncToken(String token) {
+        // Sync-Token header could have more than one value
+        final String[] syncTokens = token.split(COMMA);
+        for (final String syncTokenString : syncTokens) {
+            if (CoreUtils.isNullOrEmpty(syncTokenString)) {
+                continue;
+            }
+
+            final SyncToken syncToken;
+            try {
+                syncToken = new SyncToken(syncTokenString);
```

Comment:
Another option here, instead of throwing an exception in the constructor is using a factory pattern and returning null when `syncTokenString` isn't in the correct format. That way we don't have any overhead of stack traces when an illegal token is received